torchtune/modules/optim/cpu_offload.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torch.optim.optimizer import ParamsT
+from torchtune.config._utils import _get_component_from_path
+
+try:
+    from torchao.prototype.low_bit_optim import (
+        CPUOffloadOptimizer as _CPUOffloadOptimizer,
+    )
+
+    _CPU_OFFLOAD_OPTIM_AVAILABLE = True
+
+    class CPUOffloadOptimizer(_CPUOffloadOptimizer):
+        """Offload optimizer to CPU for single-GPU training. This will reduce GPU memory by the size of optimizer
+        state. Optimizer step will be done on CPU.
+
+        Args:
+            params (ParamsT): a list of parameters or parameter groups.
+            optimizer_class (str): name of the base optimizer.
```

Comment:
nit: explicitly say e.g. `"torch.optim.AdamW"`.
Are there any optimizers this wouldn't work with?